akka-docs/src/test/scala/docs/stream/FlowDocSpec.scala

```diff
@@ -229,3 +231,49 @@ class FlowDocSpec extends AkkaSpec {
     //#flow-async
   }
 }
+
+object FlowDocSpec {
+
+  {
+    //#materializer-from-system
+    implicit val system = ActorSystem("ExampleSystem")
+
+    implicit val mat = ActorMaterializer() // created from `system`
+    //#materializer-from-system
+  }
+
+  //#materializer-from-actor-context
+  final class RunWithMyself extends Actor {
+    implicit val mat = ActorMaterializer()
+
+    Source.maybe
+      .runWith(Sink.onComplete {
+        case Success(done) ⇒ println(s"Completed: $done")
+        case Failure(ex)   ⇒ println(s"Failed: ${ex.getMessage}")
+      })
+
+    def receive = {
+      case "boom" ⇒
+        context.stop(self) // will also terminate the stream
+    }
+  }
+  //#materializer-from-actor-context
+
+  //#materializer-from-system-in-actor
+  final class RunForever extends Actor {
+    implicit val mat = ActorMaterializer()(context.system)
```

Comment:
I'm not sure this should even be shown as an example. It's an anti-pattern. Then it's better to show passing a materializer as constructor parameter and use KillSwitch from postStop to shutdown stream that is started by the actor.

Reply:
Very true, will change